core/primitives/src/stateless_validation/stored_chunk_state_transition_data.rs

```diff
@@ -44,3 +47,20 @@ pub struct StoredChunkStateTransitionDataV2 {
     /// The code-hashes of the contracts that are deployed during the state transition.
     pub contract_deploys: Vec<CodeHash>,
 }
+
+#[derive(Debug, BorshSerialize, BorshDeserialize, ProtocolSchema)]
+pub struct StoredChunkStateTransitionDataV3 {
+    /// The partial state that is needed to apply the state transition,
+    /// whether it is a new chunk state transition or a implicit missing chunk
+    /// state transition.
+    pub base_state: PartialState,
+    /// If this is a new chunk state transition, the hash of the receipts that
+    /// were used to apply the state transition. This is redundant information,
+    /// but is used to validate against `StateChunkWitness::exact_receipts_hash`
+    /// to ease debugging of why a state witness may be incorrect.
+    pub receipts_hash: CryptoHash,
+    /// The code-hashes of the contracts that are accessed (called) during the state transition.
+    pub contract_accesses: Vec<CodeHash>,
+    /// Contracts that are deployed during the state transition.
+    pub contract_deploys: Vec<CodeBytes>,
```

Comment:
Can this field become large in size? Just double checking it won't cause any issue

Reply:
Yes it may due to code size. However, note that the partial state in this struct already contains code that was called, so this feature will save some of this space (since it will no longer contain the code that was called) but will add more space due to new deployed code.